go 1.12.1 issue on mac/Travis timeout

This is impacting almost every PR.
For example: https://travis-ci.org/GoogleContainerTools/skaffold/jobs/509855275#L53

````
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/version	1.233s	coverage: 0.5% of statements in ./pkg/..., ./cmd/...
time="2019-03-22T10:55:09Z" level=info msg="files added: [c]"
time="2019-03-22T10:55:09Z" level=info msg="files modified: [a]"
time="2019-03-22T10:55:09Z" level=info msg="files deleted: [b]"
time="2019-03-22T10:55:09Z" level=info msg="files added: [a b c]"
time="2019-03-22T10:55:09Z" level=info msg="files deleted: [a b c]"
time="2019-03-22T10:55:09Z" level=info msg="files modified: [/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/skaffold224292210/file]"
time="2019-03-22T10:55:09Z" level=info msg="files deleted: [/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/skaffold015584809/file]"
time="2019-03-22T10:55:10Z" level=info msg="files added: [/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/skaffold758672244/new]"
time="2019-03-22T10:55:10Z" level=info msg="files modified: [/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/skaffold758672244]"
PASS
SIGQUIT: quit
PC=0x7fff70a46cee m=0 sigcode=0
goroutine 0 [idle]:
runtime.pthread_cond_wait(0x6c4fb48, 0x6c4fb08, 0x0)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/runtime/sys_darwin.go:357 +0x3b
runtime.semasleep(0xffffffffffffffff, 0x60001e2148)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/runtime/os_darwin.go:63 +0x85
runtime.notesleep(0x6c4f908)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/runtime/lock_sema.go:167 +0xe0
runtime.stopm()
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/runtime/proc.go:1936 +0xc1
runtime.exitsyscall0(0xc00032c000)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/runtime/proc.go:3128 +0x118
runtime.mcall(0x405f72b)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/runtime/asm_amd64.s:299 +0x5b
goroutine 1 [runnable]:
testing.coverReport()
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/testing/cover.go:103 +0x281
testing.(*M).writeProfiles(0xc000400200)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/testing/testing.go:1296 +0x1f5
testing.(*M).after.func1()
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/testing/testing.go:1237 +0x42
sync.(*Once).Do(0xc000400260, 0xc000519d58)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/sync/once.go:44 +0xdf
testing.(*M).after(0xc000400200)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/testing/testing.go:1236 +0x62
testing.(*M).Run(0xc000400200, 0x0)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/testing/testing.go:1084 +0x531
main.main()
	_testmain.go:716 +0x345
goroutine 5 [chan receive, 2 minutes]:
github.com/GoogleContainerTools/skaffold/vendor/github.com/rjeczalik/notify.(*recursiveTree).dispatch(0xc00003c140)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/rjeczalik/notify/tree_recursive.go:125 +0x112
created by github.com/GoogleContainerTools/skaffold/vendor/github.com/rjeczalik/notify.newRecursiveTree
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/rjeczalik/notify/tree_recursive.go:119 +0x20b
goroutine 6 [syscall, 2 minutes, locked to thread]:
github.com/GoogleContainerTools/skaffold/vendor/github.com/rjeczalik/notify._Cfunc_CFRunLoopRun()
	_cgo_gotypes.go:224 +0x5f
github.com/GoogleContainerTools/skaffold/vendor/github.com/rjeczalik/notify.init.1.func1()
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/rjeczalik/notify/watcher_fsevents_cgo.go:72 +0x64
created by github.com/GoogleContainerTools/skaffold/vendor/github.com/rjeczalik/notify.init.1
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/rjeczalik/notify/watcher_fsevents_cgo.go:63 +0x5c
goroutine 19 [chan receive]:
github.com/GoogleContainerTools/skaffold/vendor/github.com/golang/glog.(*loggingT).flushDaemon(0x6c4e600)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/golang/glog/glog.go:882 +0xae
created by github.com/GoogleContainerTools/skaffold/vendor/github.com/golang/glog.init.0
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/golang/glog/glog.go:410 +0x31d
goroutine 8 [syscall, 2 minutes]:
os/signal.signal_recv(0x40618d1)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/runtime/sigqueue.go:139 +0x9f
os/signal.loop()
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/os/signal/signal_unix.go:23 +0x30
created by os/signal.init.0
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/os/signal/signal_unix.go:29 +0x4f
goroutine 27 [runnable]:
time.stopTimer(0xc0003186e8, 0xc000000001)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/runtime/time.go:120 +0x43
time.(*Ticker).Stop(...)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/time/tick.go:46
github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch.(*pollTrigger).Start.func1(0xc0003186e0, 0xc00009a8a0, 0x5cef3e0, 0xc00028da40)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch/triggers.go:85 +0x17f
created by github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch.(*pollTrigger).Start
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch/triggers.go:79 +0xcb
goroutine 52 [runnable]:
github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch.(*pollTrigger).Start.func1(0xc00003c280, 0xc0000503c0, 0x5cef3e0, 0xc0000b8100)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch/triggers.go:81 +0x141
created by github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch.(*pollTrigger).Start
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch/triggers.go:79 +0xcb
goroutine 29 [runnable]:
time.stopTimer(0xc0002d6058, 0xc000000001)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/runtime/time.go:120 +0x43
time.(*Ticker).Stop(...)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/time/tick.go:46
github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch.(*pollTrigger).Start.func1(0xc0002d6050, 0xc00009a0c0, 0x5cef3e0, 0xc00008ec40)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch/triggers.go:85 +0x17f
created by github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch.(*pollTrigger).Start
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch/triggers.go:79 +0xcb
rax    0x104
rbx    0x1f00
rcx    0x7ffeefbfe848
rdx    0x1f00
rdi    0x6c4fb48
rsi    0x200100002100
rbp    0x7ffeefbfe8e0
rsp    0x7ffeefbfe848
r8     0x0
r9     0x60
r10    0x0
r11    0x202
r12    0x6c4fb48
r13    0x7ffeefbfe868
r14    0x1
r15    0x7fffa961c340
rip    0x7fff70a46cee
rflags 0x203
cs     0x7
fs     0x0
gs     0x0
*** Test killed with quit: ran too long (2m0s).
FAIL	github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch	120.164s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/yamltags	1.232s	coverage: 1.6% of statements in ./pkg/..., ./cmd/...
ok  	github.com/GoogleContainerTools/skaffold/pkg/webhook/labels	1.388s	coverage: 0.7% of statements in ./pkg/..., ./cmd/...
make: *** [test] Error 1
The command "make test" exited with 2.

````

after some epic testing - I excluded the red herring: it is NOT the notify library that's causing the issue but something in the pollTrigger: 

https://travis-ci.org/GoogleContainerTools/skaffold/builds/510160792

````
PASS
SIGQUIT: quit
PC=0x7fff7703e8be m=0 sigcode=0
goroutine 29 [running]:
runtime: unknown pc 0x7fff7703e8be
stack: frame={sp:0x7ffeefbfe978, fp:0x0} stack=[0xc0004c6000,0xc0004c6800)
runtime: unknown pc 0x7fff7703e8be
stack: frame={sp:0x7ffeefbfe978, fp:0x0} stack=[0xc0004c6000,0xc0004c6800)
created by github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch.(*pollTrigger).Start
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch/triggers.go:79 +0xcb
goroutine 1 [runnable]:
syscall.syscall(0x10b44e0, 0x5, 0xc0004dedc0, 0x57, 0x57, 0x0, 0x0)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/runtime/sys_darwin.go:63 +0x2e
syscall.write(0x5, 0xc0004dedc0, 0x57, 0xa0, 0xc000090558, 0x106527f, 0x80000000008)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/syscall/zsyscall_darwin_amd64.go:1621 +0x93
syscall.Write(0x5, 0xc0004dedc0, 0x57, 0xa0, 0x0, 0xc0004e2d70, 0x1)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/syscall/syscall_unix.go:191 +0x6c
internal/poll.(*FD).Write(0xc000090540, 0xc0004dedc0, 0x57, 0xa0, 0x0, 0x0, 0x0)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/internal/poll/fd_unix.go:268 +0x1e0
os.(*File).write(...)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/os/file_unix.go:280
os.(*File).Write(0xc0000b20f8, 0xc0004dedc0, 0x57, 0xa0, 0x7, 0x7, 0x10421ab)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/os/file.go:145 +0xa8
fmt.Fprintf(0x2c15c20, 0xc0000b20f8, 0x29aacf5, 0x15, 0xc00051db58, 0x7, 0x7, 0x57, 0x0, 0x0)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/fmt/print.go:200 +0xcd
testing.coverReport()
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/testing/cover.go:102 +0x47f
testing.(*M).writeProfiles(0xc0003c4100)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/testing/testing.go:1296 +0x1f5
testing.(*M).after.func1()
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/testing/testing.go:1237 +0x42
sync.(*Once).Do(0xc0003c4160, 0xc00051dd58)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/sync/once.go:44 +0xdf
testing.(*M).after(0xc0003c4100)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/testing/testing.go:1236 +0x62
testing.(*M).Run(0xc0003c4100, 0x0)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/testing/testing.go:1084 +0x531
main.main()
	_testmain.go:646 +0x345
goroutine 20 [syscall, 1 minutes]:
os/signal.signal_recv(0x1097381)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/runtime/sigqueue.go:139 +0x9f
os/signal.loop()
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/os/signal/signal_unix.go:23 +0x30
created by os/signal.init.0
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/os/signal/signal_unix.go:29 +0x4f
goroutine 22 [chan receive]:
github.com/GoogleContainerTools/skaffold/vendor/github.com/golang/glog.(*loggingT).flushDaemon(0x3b73f40)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/golang/glog/glog.go:882 +0xae
created by github.com/GoogleContainerTools/skaffold/vendor/github.com/golang/glog.init.0
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/golang/glog/glog.go:410 +0x31d
goroutine 26 [runnable]:
sync.(*Mutex).Unlock(0xc000275890)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/sync/mutex.go:175 +0x130
context.(*cancelCtx).Done(0xc000275880, 0xc000000001)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/context/context.go:330 +0x95
github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch.(*pollTrigger).Start.func1(0xc0002fc5a0, 0xc0003526c0, 0x2c4dae0, 0xc000275880)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch/triggers.go:84 +0xad
created by github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch.(*pollTrigger).Start
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch/triggers.go:79 +0xcb
goroutine 49 [runnable]:
sync.(*Mutex).Unlock(0xc0003c2a50)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/sync/mutex.go:175 +0x130
context.(*cancelCtx).Done(0xc0003c2a40, 0xc000000001)
	/Users/travis/.gimme/versions/go1.12.1.darwin.amd64/src/context/context.go:330 +0x95
github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch.(*pollTrigger).Start.func1(0xc0002ca640, 0xc000051b60, 0x2c4dae0, 0xc0003c2a40)
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch/triggers.go:84 +0xad
created by github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch.(*pollTrigger).Start
	/Users/travis/gopath/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch/triggers.go:79 +0xcb
rax    0x1
rbx    0xb
rcx    0x7ffeefbfe978
rdx    0x1098ddc
rdi    0x0
rsi    0x60b3208
rbp    0x7ffeefbfe980
rsp    0x7ffeefbfe978
r8     0x1
r9     0x1
r10    0x7fff7703e8be
r11    0x246
r12    0x4fbfec0
r13    0x17e
r14    0x4fbfec8
r15    0x1
rip    0x7fff7703e8be
rflags 0x246
cs     0x7
fs     0x0
gs     0x0
*** Test killed with quit: ran too long (2m0s).
FAIL	github.com/GoogleContainerTools/skaffold/pkg/skaffold/watch	120.073s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/yamltags	1.250s	coverage: 1.7% of statements in ./pkg/...
?   	github.com/GoogleContainerTools/skaffold/pkg/webhook/constants	[no test files]
?   	github.com/GoogleContainerTools/skaffold/pkg/webhook/gcs	[no test files]
?   	github.com/GoogleContainerTools/skaffold/pkg/webhook/github	[no test files]
?   	github.com/GoogleContainerTools/skaffold/pkg/webhook/kubernetes	[no test files]
ok  	github.com/GoogleContainerTools/skaffold/pkg/webhook/labels	1.252s	coverage: 0.7% of statements in ./pkg/...
make: *** [test] Error 1
The command "make test" exited with 2.

````

--- a/pkg/skaffold/watch/triggers.go
+++ b/pkg/skaffold/watch/triggers.go
@@ -83,6 +83,7 @@ func (t *pollTrigger) Start(ctx context.Context) (<-chan bool, error) {
 				trigger <- true
 			case <-ctx.Done():
 				ticker.Stop()
+				return
 			}
 		}
 	}()

